hist/hist/test/test_TH1_SaveAs.cxx

```diff
@@ -0,0 +1,224 @@
+#include "gtest/gtest.h"
+
+#include "TString.h"
+#include "TH1.h"
+#include "TSystem.h"
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+
+struct TestSaveAs {
+   const Int_t Nbins = 5;
+   const Int_t N = Nbins+2;  // 0 and N are the under/overflow bins, resp.
+   TString fnam = "H1dump.";
+   Char_t filename[20];
+
+   void SaveHist(TString myext, TString myoption="") {
+      // Bin contents and bin errors
+      Double_t binc[N] = {5.2, 0, 10.8, 12.3, 9.5, 7.3, 15.2};
+      Double_t bine[N] = {2.1, 0,  2.5,  2.1, 3.5, 2.7,  4.7};
+
+      TH1D *h = new TH1D("h","h_title",Nbins,0,Nbins);
+      TString ytitle = "y-title";
+
+      for (int i=0; i<N; ++i) {
+         h->SetBinContent(i, binc[i]);
+         h->SetBinError(i, bine[i]);
+      }
+      std::sprintf(filename,"%s%s",fnam.Data(),myext.Data());
+      h->SaveAs(filename, myoption.Data());
+      delete h;
+      return;
+   }
+
+   bool IsGood_csv() {
+      std::sprintf(filename,"%s%s",fnam.Data(),"csv");
+      std::ifstream infile(filename, std::ios::in);
+      if (!infile) {
+         return 0;
```

Comment:
The function returns `bool` so the return value should be either `true` or `false`, i.e. better explicit than implicit.

Reply:
Ok